mypy/subtypes.py

```diff
@@ -130,24 +143,33 @@ def visit_instance(self, left: Instance) -> bool:
         if isinstance(right, TupleType) and right.fallback.type.is_enum:
             return is_subtype(left, right.fallback)
         if isinstance(right, Instance):
-            # NOTO: left.type.mro may be None in quick mode if there
+            if (left, right) in right.type.cache:
```

Comment:
Would be better style to use a method such as `if right.type.is_cached_subtype_check(left, right):`.